### DISCUSSION TITLE
Hot water cylinder with hysteresis and PID

### DISCUSSION THREAD

**Discussion by andrzej-r** [ScratMan/HASmartThermostat#163] [General]
I would like to use hysteresis control for hot water cylinder heating to minimize boiler cycling (every time water is heated boiler has to be switched to a higher flow temperature and central heating is disabled - "domestic hot water priority" control). However, I'd still like the thermostat to take _kd_ into account to minimize temperature overshoots/undershoots. While overshoots are relatively consistent, undershoots come from the rate hot water is being consumed (e.g. filling a bathtub results in larger drop than shower), so I would like the boiler to switch on as soon as sudden drop of water temperature is detected.

Is it possible to combine PID and hysteresis control, or perhaps configure PID parameters to mimic hysteresis control?

**Comment by ScratMan** [comment]
Hi, sorry but it's impossible to have both hysteresis and PID together.
You can use the minimum on and off duration settings to ensure the boiler won't cycle too much, but if the issue comes from the domestic hot water demand disabling the heating, then changing the thermostat won't be of any help.
I'm afraid the only solution would be to install a dedicated buffer + three ways mixing valve on the heating circuit to make it independent from boiler's temperature.
